### PR TITLE
Top 10 words feature part 1: parsing comments for words and updating the counts for each word in Datastore

### PR DESCRIPTION
**Pull request by apluscs**

### CODE REVIEW

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 1]
```diff
@@ -103,10 +108,44 @@ private static String convertToJsonUsingGson(Response response) {
...
+        try {
+          wordEntity.setProperty("count", (Long) wordEntity.getProperty("count") + 1);
+          datastore.put(txn, wordEntity);
+          txn.commit();
+          break;
+        } catch (Exception e) {
+          // In case transaction doesn't commit after TRANSACTION_RETRIES, just give up.
+          if (retry++ == TRANSACTION_RETRIES) {
+            txn.rollback();
+          }
```
[the remark this answers is not in the document]
Comment (acmcarther) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:146]: Consider logging in that case.

(Invisible errors are evil)

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 2]
```diff
@@ -103,10 +108,44 @@ private static String convertToJsonUsingGson(Response response) {
...
+      } catch (com.google.appengine.api.datastore.EntityNotFoundException e) {
+        // In case word is not yet in datastore...
+        wordEntity = new Entity(wordKey);
+        wordEntity.setProperty("count", 0L);
+      }
+      
+      int retry = 0;
+      while(true){
+        try {
+          wordEntity.setProperty("count", (Long) wordEntity.getProperty("count") + 1);
```
Comment (acmcarther) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:138]: Do this outside the loop.

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 3]
```diff
@@ -103,10 +108,44 @@ private static String convertToJsonUsingGson(Response response) {
...
+          datastore.put(txn, wordEntity);
```
Comment (acmcarther) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:139]: I think you might be able to do this just once, outside the loop.

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (acmcarther) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:146]: Also, if you don't rethrow here, you need to early return or else you'll try the transaction again.

[the remark this answers is not in the document]
Comment (apluscs) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:148]: Makes sense!

**Pull request merged**